server/controllers/s3.js

```diff
@@ -1,37 +1,48 @@
 const AWS = require("aws-sdk");
 
+AWS.config.update({ 
+  accessKeyId: process.env.ACCESS_KEY,
+  secretAccessKey: process.env.SECRET_ACCESS_KEY,
+  region: process.env.REGION
+});
+
 exports.uploadImage = (req, res) => {
-  if(req.file.size > 1 * 1024 * 1024) {
-    return res.status(400).json({
-      error: "Please, only upload only image upto 1Mb of size"
-    });
-  };
-  AWS.config.update({ 
-    accessKeyId: process.env.IAM_ACCESS_KEY,
-    secretAccessKey: process.env.IAM_SECRET_ACCESS_KEY,
-    region: process.env.REGION
-  });
   const s3 = new AWS.S3();
-  const params = {
-    Bucket: process.env.BUCKET_NAME,
-    Body: req.file.buffer,
-    ContentType: req.file.mimetype,
-    Key: `photo/${req.file.originalname}`
-  };
 
-  s3.upload(params, (error, data) => {
-    if (error) {
-      console.log(error);
-      return res.status(error.status).json({
-        error: `${error.message}`
-      });
-    }
+  let locationUrls = [];
+
+  const fieldNames = ['background', 'profile'];
 
-    if(data) {
-      const locationUrl = data.Location;
-      // store locationUrl to database
-    }
+  for (let index = 0; index < fieldNames.length; index++) {
+    const fieldName = fieldNames[index];
+    const file = req.files[fieldName];
+    const params = {
+      Bucket: process.env.BUCKET_NAME,
+      Body: file[0].buffer,
+      ContentType: file[0].mimetype,
+      Key: `${req.user.id}/${file[0].fieldname}`,
+    };
+    s3.upload(params, (error, data) => {
+      if (error) {
+        console.log(error);
+        return res.status(error.status).json({
+          error: `${error.message}`
+        });
+      }
+  
+      if(data) {
+        // store locationUrl in Database;
+        const locationUrl = data.Location;
+        locationUrls.push({ locationUrl, key: data.key })
+        if (locationUrls.length === fieldNames.length) {
+          return res.status(201).json({
+            locationUrls,
+            success: "File are saved successfully."
+          })

```

Comment:
I see a comment about adding it to the database but there is no persisting the location urls to the profile model. Also, what if a user wants to just upload a profile photo?  Right now on the front-end it requires them to upload a background image then a profile image. For a UX perspective this is not ideal.

Reply:
Yes, I have handle the usecase for location urls in #57 pr. 
Also, I understand that this is not a very good UX experience. Initially, I wanted to do something like [this](https://storybook.grommet.io/?path=/story/input-fileinput-multiple--multiple). But material ui doesn't support this right now. What I can do is provide a radio button to user to choose which image they want to upload. It's not very good UX, but it will decent one. What are your thoughts on this ?

Reply:
It's something that can be addressed later, I would do some further research into potential solution here. I will say, Material UI is not blocking this feature from happening. But, feel free to visit this later.

Reply:
Sure, okay.